Src/IronPython.Modules/pyexpat.cs

```diff
@@ -580,8 +582,11 @@ public class xmlparser {
 
             private MemoryStream buffer = new MemoryStream();
 
-            public void Parse(CodeContext context, string data, bool isfinal = false)
-                => Parse(context, PythonOps.MakeBytes(data), isfinal);
+            public void Parse(CodeContext context, string data, bool isfinal = false) {
+                // CPython converts to UTF-8 via PyUnicode_AsUTF8AndSize
+                if (utf8Encoding == null) utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```

Comment:
The standard UTF8Encoding from .NET defaults to 'replace' error handler, rather than 'strict'. Is it intentional here? Besides, the default .NET replacing behaviour is slightly different than CPython's (got fixed in Core 3.0). I intend to put a workaround for it in `StringOps.DoEncode`, just as it is currently done for the 'strict' case. Is there any way that encoding here can be routed through `StringOps.DoDecode`?

Reply:
Yeah, wasn't super happy with this approach, but I wanted it to be "low level". Not sure how common it is, but presumably there are other places where CPython just does a strict UTF-8 conversion so I added a helper on `StringOps` which we can always optimize later if needed.